### ISSUE TITLE
command always has prefix character "&"

### ISSUE THREAD

**Issue by QingFengpoor** [microsoft/vscode-maven#633]
**Describe the bug**
when run commands in Lifecycle or Plugins, it always starts with "&" in terminal. with CMD or BASH, syntax error, unexpected token &. 

**Environments (please complete the following information as much as possible):**
 - OS: Windows 10
 - VS Code version: 1.56.2
 - Extension version v0.30.1

**Screenshots**
xxx@xxxx MINGW64 /c/CodeProjects/Java/ReleaseNoteEditor/springmvc/releasenote issue (issue)
$ & mvn compiler:testCompile -f "c:\CodeProjects\Java\ReleaseNoteEditor\springmvc\releasenote issue\pom.xml"
bash: syntax error near unexpected token `&'

**Comment by Eskibear** [comment]
https://github.com/microsoft/vscode-maven/blob/63994716b6156fab360f64ee68148d42ecaff438/src/mavenTerminal.ts#L100-L106

That's possibly because PowerShell was wrongly detected. What terminal are you actually using?  (looks like Git Bash?) 
You can check it by command `Terminal: Select Default Profile` as below 
![image](https://user-images.githubusercontent.com/2351748/118946007-94ea3580-b988-11eb-9175-ca22b553d452.png)

**Comment by QingFengpoor** [comment]
@Eskibear  yes, output of `Terminal: select Default Profile` looks like this
```txt
PowerShell C:\WINDOWS\System32\WindowsPowerShell\v1.0\powershell.exe                               profiles
Git Bash ...
Command Prompt ...
---------------------------
Windows PowerShell ....                detected
Cygwin ....
```
by default, my terminal is Git Bash. There is something wired that when I open VSCode for this project, the terminal is Powershell, while I add the second terminal, it is Git Bash. For other folders, it is the same. Additionally, when I open the second VSCode window, the terminal is Git Bash.

**Comment by QingFengpoor** [comment]
My settings.json contains 
```json
    "terminal.integrated.tabs.enabled": true,
    "terminal.integrated.scrollback": 1600,
    "terminal.integrated.defaultProfile.windows": "Git Bash",
```
only these three conditions for terminal.

**Comment by Eskibear** [comment]
>    "terminal.integrated.defaultProfile.windows": "Git Bash",

I can reproduce your issue by specifying above setting. 

In this extension default shell is inferred from `vscode.env.shell`, defined in https://github.com/microsoft/vscode/blob/2f2b6b528d0062b81d7b3e16c88115827b84cfca/src/vs/vscode.d.ts#L8154-L8159

I believe it's a vscode issue due related to shell profiles.

**Comment by Eskibear** [comment]
See https://github.com/microsoft/vscode/issues/121760

**Comment by Eskibear** [comment]
Do you have entry "terminal.integrated.shell.windows" in your settings? If so, remove it. I just tried in latest insiders version, seems it has been fixed.

**Comment by QingFengpoor** [comment]
No. I have already noted it.
`// "terminal.integrated.shell.windows": "C:\\WINDOWS\\System32\\cmd.exe",`.

**Comment by QingFengpoor** [comment]
Hi, Eskibear. I think my situation is different from the issue mentioned above.   
Firstly, command `terminal :select default profile` shows the profile of Git Bash.
![terminal select default profile](https://user-images.githubusercontent.com/24737114/119092770-6d0ad880-ba41-11eb-936e-b1b56e423c27.png)


Secondly, I have not use "terminal.integrated.shell.<os>". All terminal settings(containing "terminal") are here:
```json
    // terminal
    "terminal.integrated.tabs.enabled": true,
    "terminal.integrated.scrollback": 1600,
    // "terminal.integrated.shell.windows": "C:\\WINDOWS\\System32\\cmd.exe",
    // "terminal.integrated.shellArgs.windows": [],
    "terminal.integrated.defaultProfile.windows": "Git Bash",
    // "terminal.integrated.automationShell.windows": "Git Bash",
```

**Comment by Eskibear** [comment]
Root cause is `vscode.env.shell` doesn't honor what you set in `terminal.integrated.defaultProfile`. I found similar issues in other popular extensions like python and code runner etc... 

It looks vscode team has fixed that (but not released into stable version), and I tried but not repro your issue in latest insiders version. Can you verify if the issue is still there in latest insiders version?  
FYI download it here: https://code.visualstudio.com/insiders/

**Comment by QingFengpoor** [comment]
Eskibear, I am out off work <img class="emoji" alt="smile" height="20" width="20" src="https://github.githubassets.com/images/icons/emoji/unicode/1f604.png"> , and my work computer is in my company. I will try it next week. Thank u.

**Comment by molly-w** [comment]
@Eskibear , I've met the same problem that launching a clean command by Maven extension throws error: `bash: syntax error near unexpected token '&'`. I test in vscode insider, but it only accepts PowerShell and Command Prompt, so i can't set gitbash as default terminal:
![image](https://user-images.githubusercontent.com/63043224/120272689-12df0280-c2e0-11eb-8c59-acc31d45c5c0.png)

**Comment by molly-w** [comment]
@Eskibear , I've met the same problem that launching a clean command by Maven extension throws error: `bash: syntax error near unexpected token '&'`. I test in vscode insider, but it only accepts PowerShell and Command Prompt, so i can't set gitbash as default terminal:
![image](https://user-images.githubusercontent.com/63043224/120272689-12df0280-c2e0-11eb-8c59-acc31d45c5c0.png)

**Comment by jaigarV** [comment]
The issue is still present in Version: 1.85.1, I have run into it while using Python and a venv. Do you have any idea when will it be released in the stable version?

**Comment by B10inform** [comment]
Hi every one. I cannot see how this issue was resolved. Was it resolved already?
After update to 1.96.4 i am getting these errors.

When i click on Run Python file i get the following error.
(base) @DESKTOP-E49B2GQ:/mnt/d/WSL2/Python/BIOPYTHON$ & D:/PYTHON/python.exe d:/WSL2/Python/BIOPYTHON/ML.py
-bash: syntax error near unexpected token `&'
(base) @DESKTOP-E49B2GQ:/mnt/d/WSL2/Python/BIOPYTHON$ 

However it works fine with python ML.py

Thank you all for the help
